syntax = "proto3";

option objc_class_prefix = "RCT";

package ringct;

service RingCT_Service {
    rpc Compute_Vote(Vote_Request) returns (Vote_Response) {}
}

message Vote_Request{
    int32 district_id = 1;
    int32 candidate_id = 2;
    int32 voter_id = 3;
}

message Vote_Response{
    // for checking if the return and req are the same destination
    int32 district_id = 1;
    int32 candidate_id = 2;
    int32 voter_id = 3;
    int32 key_image = 4;
}